src/ast/attachpoint_parser.cpp

```diff
@@ -276,13 +276,23 @@ AttachPointParser::State AttachPointParser::lex_attachpoint(
 
 AttachPointParser::State AttachPointParser::special_parser()
 {
-  // Can only have reached here if provider is `BEGIN` or `END`
-  assert(ap_->provider == "BEGIN" || ap_->provider == "END");
-
-  if (parts_.size() == 2 && parts_[1] == "*")
-    parts_.pop_back();
-  if (parts_.size() != 1) {
-    return argument_count_error(0);
+  // Can only have reached here if provider is `BEGIN` or `END` or `bpftrace`
+  assert(ap_->provider == "BEGIN" || ap_->provider == "END" ||
+         ap_->provider == "bpftrace");
+
+  if (ap_->provider == "BEGIN" || ap_->provider == "END") {
+    if (parts_.size() == 2 && parts_[1] == "*")
+      parts_.pop_back();
+    if (parts_.size() != 1) {
+      return argument_count_error(0);
+    }
+  } else if (ap_->provider == "bpftrace") {
+    if (parts_.size() != 3) {
+      return argument_count_error(2);
+    }
+    // TODO: Allow wildcard
```

Comment:
Eh, I'm not sure we should add wildcard support - and it definitely doesn't apply to BEGIN/END

Reply:
True, I'm not sure either. It would make it possible to attach a single probe to all signals received by the bpftrace runtime but I don't know if that's any useful.

Once we add the `bpftrace:keypress:...` events, wildcards could be more useful.